public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx

```diff
@@ -26,44 +26,67 @@ export interface SettingsPage {
   id: string;
   title: string;
   icon: IconName;
-  render: () => React.ReactNode;
+  component: React.ReactNode;
 }
 
-export class DashboardSettings extends PureComponent<Props> {
-  onClose = () => {
-    locationService.partial({ editview: null });
-  };
-
-  onChangePage = (editview: string) => {
-    locationService.partial({ editview });
-  };
+const updateSearchParams = (init: string, partial: string) => {
+  const searchParams = new URLSearchParams(init);
+  const newSearchParams = new URLSearchParams(partial);
+  newSearchParams.forEach((v, k) => {
+    searchParams.set(k, v);
+  });
+  return '?' + searchParams.toString();
+};
+
+const onClose = () => locationService.partial({ editview: null });
+
+const MakeEditable: React.FC<{ onMakeEditable: () => any }> = (props) => (
```

Comment:
I thought we were moving away from using `React.FC? `

Reply:
Ah right, will change.